Automattic/wp-calypso
# Move most of lib/plans to calypso-products package

#### Changes proposed in this Pull Request

As part of my goal of moving large amounts of checkout into a package outside of calypso so that it can have type safety and be used in wp-admin, I've been working to put checkout's major dependencies into packages first. Probably the biggest of those is the combination of `lib/products-values` and `lib/plans`. Previously I moved most of the dependencies of `lib/plans` into their own package (https://github.com/Automattic/wp-calypso/pull/51833). This PR moves `lib/plans` into a new package `@automattic/calypso-products`. Eventually the code in `lib/products-values` can also be moved here, but this will give us a starting place.

Note that this does not move the file `lib/plans/features-list` because that has several major dependencies on other parts of calypso (specifically `client/components/`) which are difficult to resolve right now. Eventually we should move that to a package as well, but ignoring it for now allows us to move ahead.

One of the effects of moving to a package was combining the exported constants with the exported functions; previously other code would import data either from `lib/plans/constants` or `lib/plans` (sometimes both). Now all such places import from `@automattic/calypso-products`. This may not be as semantic. I'm open to ideas about how to improve that.

This _does_ remove two exports from `lib/plans`, but they are not used anywhere as far as I can tell: 

- `PLANS_CONSTANTS_LIST`
- `getPlanObject`

#### Testing instructions

This does not change any of the functions that are moved (later commits can be made to convert them to typescript), and it attempts to update all imports of those functions. If there is any import that was missed, the calypso build should fail. Therefore automated tests should be sufficient to prove that these changes are safe, but I'd appreciate any other opinions.

(Failing style checks are expected since this does not attempt to address any pre-existing style errors that are in the touched files.)
